### ISSUE TITLE
StreamObserver.onError(Throwable t) hard to use or unintuitive

### ISSUE THREAD

**Issue by kolmodin** [grpc/grpc-java#985]
```java
public interface StreamObserver<V>  {
  void onError(Throwable t);
  // ... more methods declared.
}
```

For client calls with streaming responses, when could the error be a `Throwable` and not a `Status`? 
It seems more fitting that `onError(..)` would take a `Status` instead.

**Comment by ejona86** [comment]
@louiscryan

**Comment by ejona86** [comment]
I agree error handling can be awkward, especially with knowing what the implementation will do. `ClientCall` and `ServerCall` are much more clear in their handling. This is a consequence of being compatible with RxJava, which to those that use it is a nice win.

**Comment by louiscryan** [comment]
If we had Java8 here we could have default methods handle the trivial conversion from onError to onStatus but sadly we do not

**Comment by ejona86** [comment]
Let's consider this a doc bug to make it more clear what Throwables are generally expected to be received.

**Comment by carl-mastrangelo** [comment]
It looks like this was addressed by b0ed77ad815cddfcfc2940dc367fae8eed487ff8

Marking this as fixed.  @kolmodin If you feel this is still unclear, feel free to reopen.
